Feature/customer registration sales connector

julianhyatt opened the pull request.

sourcecube commented on bundles/availability-alert-migrator/.editorconfig:
```diff
@@ -0,0 +1,19 @@
+; This file is for unifying the coding style for different editors and IDEs.
```
Not necessary in bundle folder

sourcecube commented on bundles/availability-alert-migrator/composer.json:
```diff
@@ -0,0 +1,33 @@
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
```
7.3

sourcecube commented on bundles/availability-alert-migrator/composer.json:
```diff
@@ -0,0 +1,33 @@
+  "autoload-dev": {
+    "psr-4": {
+      "FondOfOryx\\": "tests/FondOfOryx",
+      "Generated\\": "tests/Generated"
```
src/Generated

julianhyatt committed:
:package: add CustomerRegistrationSalesConenctor bundle
:sparkles: improve CustomerRegistration bundle
- Add new plugin GenerateMissingRegistrationKeyPostStepPlugin for regenerating registration key if missing

julianhyatt committed:
wip

sourcecube commented on bundles/customer-registration/src/FondOfOryx/Zed/CustomerRegistration/Persistence/CustomerRegistrationEntityManager.php:
```diff
@@ -77,6 +77,26 @@ public function flagCustomerAsGdprAccepted(CustomerTransfer $customerTransfer):
+        $customerEntity->save();
+
+        return $customerTransfer
+            ->setRegistrationKey($customerTransfer->getRegistrationKey())
```
What is that line for, if the customerTransfer already includes the registrationKey

julianhyatt replied:
Its a fallback. Not every customer has the registration key in the db so there will be optionally generated a new one if one is missing. Can bei enabled/disabled by plugin

DerCalli reviewed (approved):
There are a few things I would format differently (long lines, empty line etc) to make the code better readable. But no reason to reject the PR.

julianhyatt committed:
wip
